### ISSUE TITLE
socket.inet_pton() passed invalid parameters in conn.py

### ISSUE THREAD

**Issue by kevinlombardo** [dpkp/kafka-python#2251]
Hello-

In conn.py, the function socket.inet_pton() is passed the variable "address". This value is sometimes a hostname rather than an IP address, causing an exception and socket.AF_UNSPEC to be returned.

According to socket documentation, this function should be passed an "ip_string" not a hostname:
socket.inet_pton(address_family, ip_string)

An example from conn.py:
Line  1462:    af = _address_family(host_and_port_str)
Line 1418:     def _address_family(address):
Line 1429:     socket.inet_pton(af, address)

**Comment by dpkp** [comment]
Yes, but a hostname is AF_UNSPEC right? Because we don't know yet whether it resolves to ipv4 or ipv6. So this is behaving as expected. If you have a suggestion for refactoring, I'm happy to look at a PR.
